README.md

```diff
@@ -7,23 +7,57 @@
 Database Stream Processor (DBSP) is a framework for computing over data streams
 that aims to be more expressive and performant than existing streaming engines.
 
-## Why DBSP?
+## DBSP Mission Statement
 
 Computing over streaming data is hard.  Streaming computations operate over
 changing inputs and must continuously update their outputs as new inputs arrive.
 They must do so in real time, processing new data and producing outputs with a
 bounded delay.
 
-Existing streaming engines have limited expressive power, which is why their use
-in the modern data stack is restricted to data preparation and visualization.
-Complex business intelligence tasks are left to traditional database systems
-where they run as periodic (e.g., daily) batch jobs.  This is inadequate in the
-world that increasingly relies on real-time alerts, recommendations, control, etc.
+We believe that software engineers and data scientists who build streaming data
+pipelines should not be exposed to this complexity.  They should be able to
+express their computations as declarative queries and use a streaming engine to
+evaluate these queries correctly and efficiently.  DBSP aims to be such an
+engine.  To this end we set the following high-level objectives:
 
-DBSP intends to change the situation by offering a simple promise: to **execute
-any query workload that can run in batch mode in streaming mode**.
+1. **Full SQL support and more.**  While SQL is just the first of potentially
+many DBSP frontends, it offers a reference point to chatacterize the
+expressiveness of the engine.  Our goal is to support the complete SQL syntax
+and semantics, including joins and aggregates, correlated subqueries, window
+functions, complex data types, time series operators, UDFs, etc.  Beyond
+standard SQL, DBSP supports recursive queries, which arise for instance in graph
+analytics problems.
 
-## Overview
+1. **Scalability in multple dimensions.**  The engine scales with the number and
+complexity of queries, streaming data rate and the amount of state the system
+maintains in order to process the queries.
+
+1. **Performance out of the box.**  The user should be able to focus on the
```

Comment:
This is a tricky one - the query plan you start from will be as important if not more important than the incrementalization. you can always write stupid plans.